src/EventListener/MultipartRequestListener.php

```diff
@@ -28,84 +27,71 @@ public function onKernelRequest(GetResponseEvent $event)
-                $length = $parsed['headers']['content-length'] ?? null;
-                $filename = $parsed['headers']['file-name'] ?? null;
-                $formName = $parsed['headers']['form-name'] ?? null;
-                $md5Sum = $parsed['headers']['content-md5'] ?? null;
+            foreach ($relatedParts as $k => $part) {
 
                 if (0 === $k) {
-                    $request->headers->add($parsed['headers']);
-
-                    if (!$mimeType) {
-                        $request->headers->remove('Content-Type');
-                    }
-
-                    if (!$length) {
-                        $request->headers->remove('Content-Length');
-                    }
-
-                    if ('application/x-www-form-urlencoded' === $mimeType) {
+                    $request->headers->add($part->getHeaders());
+                    if ('application/x-www-form-urlencoded' === $part->getHeader('Content-Type')) {
                         $output = [];
-                        parse_str($content, $output);
+                        parse_str($part->getBody(), $output);
                         $request->request->add($output);
                     } else {
-                        $this->setRequestContent($request, $content);
-                    }
-                } else {
-                    if (null !== $md5Sum && md5($content) !== strtolower($md5Sum)) {
-                        $uploadError = 'MD5';
+                        $this->setRequestContent($request, $part->getBody());
                     }
+                    continue;
+                }
 
-                    $fp = tmpfile();
-                    fwrite($fp, $content);
-                    rewind($fp);
+                $contentDisposition = $part->getHeader('Content-Disposition');
 
-                    if ($filename) {
-                        $tmpPath = stream_get_meta_data($fp)['uri'];
-                        $file = new UploadedFile($tmpPath, $filename, $mimeType, $length ?: strlen($content), $uploadError ?? null, true);
-                        @$file->ref = $fp;
+                if ($contentDisposition === null) {
+                    continue;
+                }
 
-                        if (isset($formName)) {
-                            $formPath = $this->parseKey($formName);
+                $fp = tmpfile();
+                fwrite($fp, $part->getBody());
+                rewind($fp);
 
-                            $files = $request->files->all();
-                            $files = $this->mergeFilesArray($files, $formPath, $file);
-                            $request->files->replace($files);
-                        } else {
-                            $attachments[] = $file;
-                        }
-                    } elseif (!isset($uploadError)) {
-                        $relatedParts[] = new RelatedPart($fp, $parsed['headers']);
-                    }
+                $tmpPath = stream_get_meta_data($fp)['uri'];
```

Comment:
Would be nice to fina a way to have "file upload" but without having to create temporary files on the disk...